lib/rouge/lexers/kotlin.rb

```diff
@@ -29,6 +29,8 @@ class Kotlin < RegexLexer
 
       id = %r'(#{name_backtick})'
 
+      upper = /[\p{Lu}$_]/
```

Comment:
It seems weird to me that `$` and `_` would be part of a regular expression called 'upper'. What's the logic here?

Reply:
I copied it directly from the Scala lexer that you linked in the issue. Will give it a review, I just took it directly as it was already being used in that lexer.

Reply:
Ha! There you go. I confess when I wrote that I'd only been a maintainer for a short time and wasn't as observant as I (hopefully) am now :) I don't know Scala well enough to know what reason this would have been done but my initial reaction is that it looks unnecessary (at least in the Kotlin context).

Reply:
Yeah likewise, I was surprised it worked. Will have a look in more detail.

Reply:
My layman's understanding of Scala is that they mash in a lot of concepts so perhaps the concept of 'class' is larger than it is in Java/Kotlin.

Reply:
$ and _ not required.